### ISSUE TITLE
Releases (Heroku) issue resolving faulty

### ISSUE THREAD

**Issue by sfcgeorge** [getsentry/sentry#5588]
Using Sentry web interface, with Heroku releases set up but no code repository connected.

When clicking "Resolve in the next release" is actually sets it to resolved in the current release. Clicking "Resolve in the current release" does exactly the same thing.

Once you have clicked "Resolve in current release" you get the message "This issue has been marked as resolved as of version a2df621", fine, but when clicking on and viewing that release it shows "Issues Resolved in this Release: No issues resolved"

Basically nothing about releases linked with issue tracking seems to work properly and it never has for us for months ¯⁠\_(ツ)_/⁠¯ Maybe just an issue with Heroku releases, or because we don't have a linked repository. Dunno.

**Comment by dcramer** [comment]
I was typing up a reply and I actually think you're right in this might be broken. Here's the intention:

- This is expected "resolved in the next release" so if it happens in that release, but not in the current release, we consider it broken.
- This is "resolved in the current release" suggests its resolved right now, as of the current version, and if it happens in the current version then we know thats not true.

I'm going to dig into this today, but the current release/explicit release are newer features, and "resolved in next release" definitely works correctly.

**Comment by dcramer** [comment]
I think the best course of action here is to add a couple of things:

- An attribute on GroupResolution that indicates whether its inclusive or exclusion of the release. For example, "In Next Release" is exclusive, and "In X Release" is inclusive. We might want to make this specific to saying something closer to what the feature is.
- Create rows in GroupResolution for GroupCommitResolution? We could then augment the "Issues resolved in release" with all of the items.

**Comment by dcramer** [comment]
Decided we're going to add GroupResolution.type, and contain the logic in there.

**Comment by sfcgeorge** [comment]
Thank you for the fast feedback and I'm amazed by the speedy fix already.

Now I think "resolved in current / upcoming" release does exactly what I expected. Useful for long CI builds when you just want to tick off an issue and move on but the release isn't there yet, then once it goes live the release gets filled in on the issue, perfect.

No worries about the upcoming features then, I look forward to seeing how it comes out. I was just playing with buttons really :)

I'm glad we chose Sentry over any of the other error trackers we tries!
